networkx/algorithms/operators/product.py

```diff
@@ -532,3 +533,85 @@ def corona_product(G, H):
         GH.add_edges_from((G_node, (G_node, H_node)) for H_node in H)
 
     return GH
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+@nx._dispatchable(graphs=_G_H)
+def modular_product(G, H):
+    r"""Returns the Modular product of G and H.
+
+    The modular product of $G$ and $H$ is the graph $M = G \nabla H$,
+    consisting of the node set $V(M) = V(G) \times V(H)$ that is the Cartesian
+    product of the node sets of $G$ and $H$. Further, $M$ contains an edge $((u, v), (x, y))$:
+
+    - if $u$ is adjacent to $x$ in $G$ and $v$ is adjacent to $y$ in $H$, or
+    - if $u$ is not adjacent to $x$ in $G$ and $v$ is not adjacent to $y$ in $H$.
+
+    More formally, $E(M) = \{((u, v), (x, y)) \mid (u, x) \in E(G) \text{ and } (v, y) \in E(H)\}
+    \cup \{((u, v), (x, y)) \mid (u, x) \notin E(G) \text{ and } (v, y) \notin E(H)\}$
```

Comment:
Sometimes the [latex doesn't work well in sphinx](https://output.circle-artifacts.com/output/job/e71d889b-c7bd-4ec9-a92f-6fb49d91a3b7/artifacts/0/doc/build/html/reference/algorithms/generated/networkx.algorithms.operators.product.modular_product.html#networkx.algorithms.operators.product.modular_product). If it is mostly text-like we can make do with a literal block
```suggestion
    More formally::

        E(M) = {((u, v), (x, y)) | ((u, x) in E(G) and (v, y) in E(H)) or 
                                               ((u, x) not in E(G) and (v, y) not in E(H))}

```

Reply:
Good idea, looks more clear as well!